cmd/nerdctl/container/container_run_cgroup_linux_test.go

```diff
@@ -170,6 +174,53 @@ func TestRunCgroupV1(t *testing.T) {
 	base.Cmd("run", "--rm", "--cpu-quota", "42000", "--cpu-period", "100000", "--cpuset-mems", "0", "--memory", "42m", "--memory-reservation", "6m", "--memory-swap", "100m", "--memory-swappiness", "0", "--pids-limit", "42", "--cpu-shares", "2000", "--cpuset-cpus", "0-1", testutil.AlpineImage, "cat", quota, period, cpusetMems, memoryLimit, memoryReservation, memorySwap, memorySwappiness, pidsLimit, cpuShare, cpusetCpus).AssertOutExactly(expected)
 }
 
+// TestIssue3781 tests https://github.com/containerd/nerdctl/issues/3781
+func TestIssue3781(t *testing.T) {
+	t.Parallel()
+	testCase := nerdtest.Setup()
+	testCase.Require = test.Not(nerdtest.Docker)
```

Comment:
?

Reply:
because i will use containerd client to get container id and use spec, err := container.Spec(ctx) to get spec, this may failed with docker.

Reply:
Rather than depending on `client.LoadContainer`, can we just exec `cat /sys/fs/cgroup/pids.max` in the container?

Reply:
old nerdctl  cat /sys/fs/cgroup/pids.max is right if use goshim.  runc will ignore pids.max=0  see:https://github.com/opencontainers/runc/issues/4566
It only effect rustshim.  https://github.com/containerd/rust-extensions.